apple/validator.go

```diff
@@ -65,6 +65,24 @@ func NewWithURL(validationURL string, revokeURL string) *Client {
 	return client
 }
 
+// NewWithClient creates a Client object with a custom http client
+func NewWithClient(httpClient *http.Client) *Client {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
```

Comment:
This logic feels like it is adding hidden functionality that the user may not expect. If I as a developer call `NewWithClient` without a client (by accident), then this would sub in a default client and I may be wondering why my own client isn't being used. 

We may be better turning this into an error. If the user wants to use the default client, they can always call `New()` instead. Instead, we should surface an error to the user. 

```suggestion
	if httpClient == nil {
		return nil, fmt.Errorf("provided httpClient is nil")
	}
```

Just curious, is there a scenario you are running into where sometimes the httpclient being passed in is nil and sometimes it is not?